Documentation/ceph-best-practices.md

```diff
@@ -0,0 +1,516 @@
+---
+title: BestPractices
+weight: 3950
+indent: true
+---
+
+# Rook-Ceph Best Practices
+
+## Overview
+
+Ceph and Kubernetes are both complex tools and harmonizing the interactions between the two can be
+daunting. This is especially true for users who are new to operating either system, prompting
+questions such as:
+ * How can I restrict Ceph to a portion of my nodes?
+ * Can I set Kubernetes CPU or RAM limits for my Ceph daemons?
+ * What are some ways to get better performance from my cluster?
+
+This document covers tested patterns and best practices to answer these questions and more. Our
+examples will help you configure and manage your Ceph cluster running in Kubernetes to meet your
+needs. The following examples and advice are based on Ceph Octopus (v15) with Rook v1.3 running in a
```

Comment:
I would push back very hard on this. I think this is an important piece of information, and if we make changes to this doc with newer versions of k8s, we should re-verify all of these recommendations. Otherwise, there is no real quality to what is here since none of this is tested in CI. If we can start testing best practices in CI, then I think your point is very valid.

Reply:
I see best practices as principles that don't change with versions. The implementation details can certainly change. If we treat this as an "Advanced Walkthrough" with a very specific tested configuration, then it makes sense to mention the version where it's tested. This is another reason to extract general best practices into other docs. We shouldn't have to re-validate all the best practices when one of them changes.